cql3/restrictions/single_column_restriction.hh

```diff
@@ -340,6 +340,7 @@ public:
         }
 
```

Comment:
I think your calls to std::max_element() are not needed due to this. It ensures that we have at most one upper bound and at most one lower bound.

However, no need to remove them, it's more general and allows us to relax needless restrictions around CQL, making @kostja happy(ier).